packages/block-editor/src/components/media-upload/index.native.js

```diff
@@ -147,6 +154,18 @@ export class MediaUpload extends Component {
 							source.types.includes( allowedType )
 					  );
 			} )
+			.filter( ( source ) => {
+				if (
+					allowedTypes.every(
+						( allowedType ) => allowedType === MEDIA_TYPE_AUDIO
+					)
+				) {
+					if ( source.id !== URL_MEDIA_SOURCE ) {
```

Comment:
I think we could get rid of this `if` block if we move this condition to the above one:

```
				if (
					allowedTypes.every(
						( allowedType ) => allowedType === MEDIA_TYPE_AUDIO
					) &&
					source.id !== URL_MEDIA_SOURCE
				) {
					return isAudioBlockEnabled === true;
				}
```

Reply:
Resolved logic related to these suggestions here cec0331 245a895